audit/audit-template.yml

```diff
@@ -178,6 +179,18 @@ Resources:
             Transitions:
               - TransitionInDays: 90
                 StorageClass: GLACIER
+      NotificationConfiguration:
+        !If
+          - IsProduction
+          -
+            QueueConfigurations:
+              - Event: "s3:ObjectCreated:*"
+                Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
+              - Event: "s3:ObjectRemoved:*"
+                Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
+              - Event: "s3:ObjectRestore:*"
+                Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
```

Comment:
AFAIK we have only ever dealt with `s3:ObjectCreated` notifications. `s3:ObjectRestore` might work. Notifying on `s3:ObjectRemoved` doesn't seem like a good idea though, since the S3 processor tries to retrieve the S3 object referred to in the notification. Looking at the CSLS source code:
```python
            for record in lambda_event["Records"]:
                sqs_body = json.loads(record["body"])
                for message in sqs_body["Records"]:
                    ...
                    s3_object = s3_get_object(bucket, key)
                    if s3_object:
                        ...
                    else:
                        LOG.error("Unable to retrieve S3 object")
````
`s3_get_object()` wraps a `GetObject` request in a try/except statement, so the whole lambda shouldn't fall over if the object doesn't exist (which would be the case for `s3:ObjectRemoved` notifications, I assume). However I think it would be better for various reasons (such as performance of the lambda) if it wasn't notified for events that no longer exist. If you want to delete events in Splunk itself, I'm afraid CSLS can't do that.

tl;dr: You shouldn't notify on `s3:ObjectRemoved`